recipe/bld.bat

```diff
@@ -1,21 +1,14 @@
 REM edited from the one from Anaconda
 
 REM Build step
-nmake -f makefile.msc
-jom -f makefile.msc lib bzip2
-if errorlevel 1 exit 1
+nmake -f makefile.msc lib || exit 1
 
 REM Install step
-copy libbz2.lib %LIBRARY_LIB%\libbz2_static.lib || exit 1
-REM Some packages expect 'bzip2.lib', so make another copy
-copy libbz2.lib %LIBRARY_LIB%\bzip2_static.lib || exit 1
 copy bzlib.h %LIBRARY_INC% || exit 1
-
-cl /MD /O2 /Ibzlib.h /Ibzlib_private.h /D_USRDLL /D_WINDLL blocksort.c bzlib.c compress.c crctable.c decompress.c huffman.c randtable.c /LD /Felibbz2.dll /link /DEF:libbz2.def
-if errorlevel 1 exit 1
-
+copy libbz2_static.lib %LIBRARY_LIB% || exit 1
 copy libbz2.lib %LIBRARY_LIB% || exit 1
+copy libbz2.dll %LIBRARY_BIN% || exit 1
+
 REM Some packages expect 'bzip2.lib', so make copies
+copy libbz2_static.lib %LIBRARY_LIB%\bzip2_static.lib || exit 1
 copy libbz2.lib %LIBRARY_LIB%\bzip2.lib || exit 1
-copy libbz2.dll %LIBRARY_BIN% || exit 1
-copy libbz2.dll %LIBRARY_BIN%\bzip2.dll || exit 1
```

Comment:
FYI this removal of `%LIBRARY_BIN\bzip2.dll` creates a bit of inconsistency with other Windows packages, e.g. hdf5, lzo2, lzo1 blosc which are all accessible via the non-`lib-`-prefixed versions in the `%LIBRARY_BIN%`, e.g.:

https://github.com/conda-forge/lzo-feedstock/blob/75ffb613aeaab06b5bef5121f0efcfcb75329a32/recipe/meta.yaml#L34

FWIW this [broke a build I had going over in PyTables](https://github.com/PyTables/PyTables/pull/1183#issuecomment-2229020526). So might be worth reverting this in case other people look specifically for the windows-style naming as well?